wgengine/magicsock/dontfrag_darwin.go

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package magicsock
+
+import (
+	"net"
+	"syscall"
+
+	"tailscale.com/types/nettype"
+)
+
+const (
+	// From https://opensource.apple.com/source/xnu/xnu-6153.141.1/bsd/netinet6/in6.h.auto.html
+	socketOptionIPDontFrag   = 28
+	socketOptionIPv6DontFrag = 62
```

Comment:
So I finally looked up why we have to do this and IPv6 still automatically fragments packets sent on UDP and raw sockets! I guess it's just TCP packets that don't get fragmented on IPv6. So maybe expand this comment to add the below link and the above fact. And maybe a TODO on getting these into the golang sockopt library?

https://datatracker.ietf.org/doc/html/rfc3542#section-11.2

Reply:
Yeah, the subtlety you probably are aware of is that UDP datagrams never had a traditional association with MTU, unlike the MSS exchange with TCP, so fragmentation is/was a necessary evil to avoid changing in-flow semantics in the userspace APIs.

I'm not fully up to date on the IPv6 RFC's but I'd hazard a guess that fragmentation is coming back for v6 TCP as well as part of the MPTCP work - just a guess though.

Reply:
There's a subtle drop-down on pkg.go.dev that adds ?GOOS=darwin to the url and renders that goos